src/StreamJsonRpc/Reflection/RpcMarshalableKnownSubtypeAttribute.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace StreamJsonRpc;
+
+/// <summary>
+/// Attribute to be used on an interface marked with <see cref="RpcMarshalableAttribute"/> to indicate that a known
+/// inherited interface exists.
+///
+/// When an object that implements <see cref="SubType"/> is marshaled as its base interface marked with
+/// <see cref="RpcMarshalableKnownSubtypeAttribute"/>, <see cref="SubTypeCode"/> is included in the StreamJsonRpc
+/// message allowing the creation of a proxy which implements <see cref="SubType"/>.
+/// </summary>
+/// <remarks>
+/// If an object implements more than one of the <see cref="SubType"/> insterfaces, the behavior is undefined.
+///
+/// In an object doesn't implement any of the <see cref="SubType"/> interfaces,
+/// <see cref="RpcMarshalableKnownSubtypeAttribute"/> attributes are simply ignored.
+/// </remarks>
+[AttributeUsage(AttributeTargets.Interface, AllowMultiple = true, Inherited = false)]
+public class RpcMarshalableKnownSubtypeAttribute : Attribute
```

Comment:
nit: you use `SubType` casing in some places an `Subtype` in others. We should be consistent. I'm leaning toward `SubType`, but if "subtype" is a word, then `Subtype` is also acceptable.